### ISSUE TITLE
Danger Plugin - that tells you what classes are too fat 

### ISSUE THREAD

**Issue by orta** [orta/life#19]
Let's take this

`find . -path './Pods' -prune -o -name "*.m" -exec wc -l "{}" \; | sort -n`

and this post by @chriseidhof on [How to keep your classes small and clean](http://chris.eidhof.nl/post/43925321331/how-to-keep-your-classes-small-and-clean).

and turn it into something that you can have as a part of your CI tooling.

The name could be related to having less chocolate in your diet, or about [fat controllers](https://encrypted.google.com/search?tbm=isch&q=fat%20controller&tbs=imgo:1#facrc=0%3Bfat%20controller%20meme&imgrc=_) - could be nice to play with that metaphor in errors  & warnings etc.

```
~/s/i/energy (toolbar_polish ↩) $ diet --max=550

      ✗  KKGridView.m ( 1469 ) 
      ✗  RESTKit/NSManagedObject+ActiveRecord.m ( 788 )  
      ✗  Views/ARGridView.m ( 610 )  
      ✗  Controllers/ARArtworkSetViewController.m ( 609 )  
      ✗  Controllers/AREmailArtworksViewController.m ( 524 )  
      ✓  Util/ARArtworkInfoViewController.m ( 517 )  
      ✓  CoreTextWrapper/Views/AKOMultiColumnTextView.m ( 479 )  

Failing due to 6 files being above the limit.

```

| Option name | Does this |
| :-- | :-- |
| type | allow only view controllers (e.g. only look at things with ViewController in name ) |
| skip | skip specific folders |
| max | maximum number of LOC |
| warn | don't do an error return code if something is over XXX LOC |

**Comment by orta** [comment]
Maybe call it `xcdiet` to fir the general naming scheme

**Comment by orta** [comment]
Maybe wrap [sloccount](http://www.dwheeler.com/sloccount/) - from [this tweet](@orta @chriseidhof sloccount can report size of files; comes with plugin for Jenkins, but can't fail build https://t.co/DBmZy0a2xX).
Perhaps have it skip the preceding lines that begin with `#` so that it doesn't pollute the return value.

**Comment by chriseidhof** [comment]
This is an awesome idea.

## 

Chris Eidhof

**Comment by fabiopelosin** [comment]
What about forking RuboCop into something like CocoCop?

**Comment by orta** [comment]
`0.0` sounds like an epic task.
ATM my hopes for that rest on [OCLint](http://oclint.org).

**Comment by fabiopelosin** [comment]
> 0.0 sounds like an epic task.

CocoaPods style :wink:

**Comment by orta** [comment]
also, should be able to skip empty lines in the LOC count

**Comment by Lascorbe** [comment]
Did you know about CLOC? http://cloc.sourceforge.net :smile:

**Comment by orta** [comment]
Yeah, perhaps a wrapper with defaults around CLOC might be enough TBH.

**Comment by orta** [comment]
Could probably be built into Danger as a plugin
